src/components/avatar/Avatar.tsx

```diff
@@ -76,47 +70,47 @@ export const Avatar = (props: AvatarProps) => {
     [`Avatar-content--${AvatarAppearance}`]: AvatarAppearance,
   });
 
-  const renderAvatar = () => {
-    return (
-      <span data-test="DesignSystem-Avatar" {...baseProps} className={classes}>
-        {initials && (
-          <Text weight="medium" appearance={'white'} className={ContentClass}>
-            {initials}
-          </Text>
-        )}
-        {!initials && (
-          <Icon
-            data-test="DesignSystem-Avatar--Icon"
-            name="person"
-            size={size === 'regular' ? 16 : 12}
-            appearance={'white'}
-            className={IconClass}
-          />
-        )}
-      </span>
-    );
-  };
-
-  const renderTooltip = () => {
-    if (withTooltip && initials) {
-      return (
-        <>{renderAvatar()}</>
-        // <Tooltip tooltip={tooltip} position={tooltipPosition} triggerClass={'flex-grow-0'}>
-        // </Tooltip>
-      );
-    }
-
-    return renderAvatar();
+  const imageDimensions = {
+    width: `${size === 'regular' ? 32 : 24}px`,
+    height: `${size === 'regular' ? 32 : 24}px`,
   };
 
-  return renderTooltip();
+  return (
+    <span title={tooltip} data-test="DesignSystem-Avatar" {...baseProps} {...rest} className={classes}>
+      {text && (
+        <Text weight="medium" appearance={'white'} className={ContentClass}>
+          {text}
+        </Text>
+      )}
+      {icon && (
+        <Icon
+          data-test="DesignSystem-Avatar--Icon"
+          name={icon}
+          size={size === 'regular' ? 16 : 12}
+          appearance={'white'}
+          className={IconClass}
+        />
+      )}
+      {image?.src && (
+        <img data-test="DesignSystem-Avatar--Image" src={image.src} alt={image.altText} {...imageDimensions} />
+      )}
+      {!text && !image?.src && !icon && (
+        <Icon
+          data-test="DesignSystem-Avatar--Icon"
+          name="person"
+          size={size === 'regular' ? 16 : 12}
+          appearance={'white'}
+          className={IconClass}
+        />
+      )}
+    </span>
+  );
 };
 
 Avatar.displayName = 'Avatar';
 Avatar.defaultProps = {
-  tooltipPosition: 'bottom',
-  withTooltip: true,
```

Comment:
We need to add support to handle tooltip position

Reply:
Since we are wrapping with Tooltip we don't need to support it in Avatar